src/radical/pilot/agent/scheduler/base.py

```diff
@@ -653,18 +653,29 @@ def _schedule_waitpool(self):
         # We define `tuple_size` as
         #     `(cpu_processes + gpu_processes) * cpu_threads`
         #
-        tasks = list(self._waitpool.values())
-        tasks.sort(key=lambda x:
+        tasks   = list(self._waitpool.values())
+        to_wait = list()
+        to_test = list()
+
+        for task in tasks:
+            named_env = task['description'].get('named_env')
+            if named_env:
+                if os.path.exists('%s.ok' % named_env):
+                    to_test.append(task)
+                else:
+                    to_wait.append(task)
```

Comment:
need to limit IO calls, thus to have corresponding flags for `named_env`
as an example:
```python
        named_envs = {}
        for task in tasks:
            named_env = task['description'].get('named_env')
            if named_env:
                if named_env not in named_envs:
                    named_envs[named_env] = os.path.exists('%s.ok' % named_env)
                if named_envs[named_env]: to_test.append(task)
                else                    : to_wait.append(task)
```